x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -157,7 +174,13 @@ public DownsampleIndexerAction.ShardDownsampleResponse execute() throws IOExcept
         );
 
         if (task.getNumIndexed() != task.getNumSent()) {
-            task.setRollupShardIndexerStatus(RollupShardIndexerStatus.FAILED);
+            task.updatePersistentTaskState(
```

Comment:
maybe we should update the `task.updatePersistentTaskState(...)` in `RollupShardPersistentTaskExecutor#nodeOperation(...)`? When can just catch the exceptions thrown here there and set the status to failed.